tests/test_when.py

```diff
@@ -11,6 +11,10 @@
 from magma.backend.mlir.errors import MlirWhenCycleError
 
 
+# Change this to validate mlir->verilog compile doesn't error
+_OUTPUT_TYPE = "mlir"
```

Comment:
not sure what the purpose of this is? we should either be testing mlir or mlir-verilog. if we need to test the latter then it should just be that

Reply:
I use it locally to ensure the compilation doesn't fail, this allows me to change the output for the entire file with a single variable, but we leave it on mlir for CI for speed

Reply:
ok, let's make it into a structured environment variable then